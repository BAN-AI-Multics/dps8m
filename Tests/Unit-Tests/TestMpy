load TestMpy.oct
;set debug stdout
;set cpu debug=trace;regdumpaqi
g 0
q
