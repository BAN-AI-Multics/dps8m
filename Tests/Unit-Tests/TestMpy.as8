	equ	printaq, 19
	equ	puts, 16
	equ	halt, 18

	org	0

	ldq	=0
	mpy	=0
	emcall	printaq
	eaa	msg0
	emcall	puts

	ldq	wee
	mpy	wee
	emcall	printaq
	eaa	msgbig
	emcall	puts

	ldq	=1
	mpy	=10
	emcall	printaq
	eaa	msg10
	emcall	puts

	mpy	=10
	emcall	printaq
	eaa	msg100
	emcall	puts

	mpy	=10
	emcall	printaq
	eaa	msg1000
	emcall	puts

	mpy	=10
	emcall	printaq
	eaa	msg10000
	emcall	puts

	mpy	=10
	emcall	printaq
	eaa	msg100000
	emcall	puts

	mpy	=10
	emcall	printaq
	eaa	msg1000000
	emcall	puts

	mpy	=10
	emcall	printaq
	eaa	msg10000000
	emcall	puts

	mpy	=10
	emcall	printaq
	eaa	msg100000000
	emcall	puts

	mpy	=10
	emcall	printaq
	eaa	msg1000000000
	emcall	puts

	mpy	=10
	emcall	printaq
	eaa	msg10000000000
	emcall	puts

	mpy	=10
	emcall	printaq
	eaa	msg100000000000
	emcall	puts

	emcall	halt

wee:	oct	400000000000
one:	dec	1
ten:	dec	10
msg0:	aci	' (0)\r\n0'
msgbig:	aci	' (1180591620717411303424)\r\n\0'
msg10:	aci	' (10)\r\n\0'
msg100:	aci	' (100) \r\n\0'
msg1000:	aci	' (1000)\r\n\0'
msg10000:	aci	' (10000)\r\n\0'
msg100000:	aci	' (100000)\r\n\0'
msg1000000:	aci	' (1000000) \r\n\0'
msg10000000:	aci	' (10000000)\r\n\0'
msg100000000:	aci	' (100000000)\r\n\0'
msg1000000000:	aci	' (1000000000)\r\n\0'
msg10000000000:	aci	' (10000000000) \r\n\0'
msg100000000000:	aci	' (100000000000)\r\n\0'
