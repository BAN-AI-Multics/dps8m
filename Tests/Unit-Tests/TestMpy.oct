!SIZE 000172
000000 xxxx 000154236000 	ldq	=0
000001 xxxx 000155402000 	mpy	=0
000002 xxxx 000023420400 	emcall	printaq
000003 xxxx 000073635000 	eaa	msg0
000004 xxxx 000020420400 	emcall	puts
000005 xxxx 000070236000 	ldq	wee
000006 xxxx 000070402000 	mpy	wee
000007 xxxx 000023420400 	emcall	printaq
000010 xxxx 000075635000 	eaa	msgbig
000011 xxxx 000020420400 	emcall	puts
000012 xxxx 000156236000 	ldq	=1
000013 xxxx 000157402000 	mpy	=10
000014 xxxx 000023420400 	emcall	printaq
000015 xxxx 000104635000 	eaa	msg10
000016 xxxx 000020420400 	emcall	puts
000017 xxxx 000160402000 	mpy	=10
000020 xxxx 000023420400 	emcall	printaq
000021 xxxx 000106635000 	eaa	msg100
000022 xxxx 000020420400 	emcall	puts
000023 xxxx 000161402000 	mpy	=10
000024 xxxx 000023420400 	emcall	printaq
000025 xxxx 000111635000 	eaa	msg1000
000026 xxxx 000020420400 	emcall	puts
000027 xxxx 000162402000 	mpy	=10
000030 xxxx 000023420400 	emcall	printaq
000031 xxxx 000114635000 	eaa	msg10000
000032 xxxx 000020420400 	emcall	puts
000033 xxxx 000163402000 	mpy	=10
000034 xxxx 000023420400 	emcall	printaq
000035 xxxx 000117635000 	eaa	msg100000
000036 xxxx 000020420400 	emcall	puts
000037 xxxx 000164402000 	mpy	=10
000040 xxxx 000023420400 	emcall	printaq
000041 xxxx 000122635000 	eaa	msg1000000
000042 xxxx 000020420400 	emcall	puts
000043 xxxx 000165402000 	mpy	=10
000044 xxxx 000023420400 	emcall	printaq
000045 xxxx 000126635000 	eaa	msg10000000
000046 xxxx 000020420400 	emcall	puts
000047 xxxx 000166402000 	mpy	=10
000050 xxxx 000023420400 	emcall	printaq
000051 xxxx 000132635000 	eaa	msg100000000
000052 xxxx 000020420400 	emcall	puts
000053 xxxx 000167402000 	mpy	=10
000054 xxxx 000023420400 	emcall	printaq
000055 xxxx 000136635000 	eaa	msg1000000000
000056 xxxx 000020420400 	emcall	puts
000057 xxxx 000170402000 	mpy	=10
000060 xxxx 000023420400 	emcall	printaq
000061 xxxx 000142635000 	eaa	msg10000000000
000062 xxxx 000020420400 	emcall	puts
000063 xxxx 000171402000 	mpy	=10
000064 xxxx 000023420400 	emcall	printaq
000065 xxxx 000147635000 	eaa	msg100000000000
000066 xxxx 000020420400 	emcall	puts
000067 xxxx 000022420400 	emcall	halt
000070 xxxx 400000000000 wee:	oct	400000000000
000071 xxxx 000000000001 one:	dec	1
000072 xxxx 000000000012 ten:	dec	10
000073 xxxx 040050060051 msg0:	aci	' (0)\r\n0'
000074 xxxx 015012060000 
000075 xxxx 040050061061 msgbig:	aci	' (1180591620717411303424)\r\n\0'
000076 xxxx 070060065071 
000077 xxxx 061066062060 
000100 xxxx 067061067064 
000101 xxxx 061061063060 
000102 xxxx 063064062064 
000103 xxxx 051015012000 
000104 xxxx 040050061060 msg10:	aci	' (10)\r\n\0'
000105 xxxx 051015012000 
000106 xxxx 040050061060 msg100:	aci	' (100) \r\n\0'
000107 xxxx 060051040015 
000110 xxxx 012000000000 
000111 xxxx 040050061060 msg1000:	aci	' (1000)\r\n\0'
000112 xxxx 060060051015 
000113 xxxx 012000000000 
000114 xxxx 040050061060 msg10000:	aci	' (10000)\r\n\0'
000115 xxxx 060060060051 
000116 xxxx 015012000000 
000117 xxxx 040050061060 msg100000:	aci	' (100000)\r\n\0'
000120 xxxx 060060060060 
000121 xxxx 051015012000 
000122 xxxx 040050061060 msg1000000:	aci	' (1000000) \r\n\0'
000123 xxxx 060060060060 
000124 xxxx 060051040015 
000125 xxxx 012000000000 
000126 xxxx 040050061060 msg10000000:	aci	' (10000000)\r\n\0'
000127 xxxx 060060060060 
000130 xxxx 060060051015 
000131 xxxx 012000000000 
000132 xxxx 040050061060 msg100000000:	aci	' (100000000)\r\n\0'
000133 xxxx 060060060060 
000134 xxxx 060060060051 
000135 xxxx 015012000000 
000136 xxxx 040050061060 msg1000000000:	aci	' (1000000000)\r\n\0'
000137 xxxx 060060060060 
000140 xxxx 060060060060 
000141 xxxx 051015012000 
000142 xxxx 040050061060 msg10000000000:	aci	' (10000000000) \r\n\0'
000143 xxxx 060060060060 
000144 xxxx 060060060060 
000145 xxxx 060051040015 
000146 xxxx 012000000000 
000147 xxxx 040050061060 msg100000000000:	aci	' (100000000000)\r\n\0'
000150 xxxx 060060060060 
000151 xxxx 060060060060 
000152 xxxx 060060051015 
000153 xxxx 012000000000 
000154 xxxx 000000000000 decimal literal (=0)
000155 xxxx 000000000000 decimal literal (=0)
000156 xxxx 000000000001 decimal literal (=1)
000157 xxxx 000000000012 decimal literal (=10)
000160 xxxx 000000000012 decimal literal (=10)
000161 xxxx 000000000012 decimal literal (=10)
000162 xxxx 000000000012 decimal literal (=10)
000163 xxxx 000000000012 decimal literal (=10)
000164 xxxx 000000000012 decimal literal (=10)
000165 xxxx 000000000012 decimal literal (=10)
000166 xxxx 000000000012 decimal literal (=10)
000167 xxxx 000000000012 decimal literal (=10)
000170 xxxx 000000000012 decimal literal (=10)
000171 xxxx 000000000012 decimal literal (=10)
