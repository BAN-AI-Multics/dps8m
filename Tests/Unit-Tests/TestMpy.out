
dps-8/m simulator V4.0-0 Beta        git commit id: c8ae76cb
122 (000172) words loaded
0 (0)
01180591620717411303424 (1180591620717411303424)
10 (10)
100 (100) 
1000 (1000)
10000 (10000)
100000 (100000)
1000000 (1000000) 
10000000 (10000000)
100000000 (100000000)
1000000000 (1000000000)
10000000000 (10000000000) 
100000000000 (100000000000)

simCycles = 112

cpuCycles = 56
TestMpy-4> g 0

Halt, IC: 000067 (000022420400)
Goodbye
